### PR TITLE
Rel constants

### PR DESCRIPTION
**Pull request by Crell**
Supersedes #2.

I like the idea of adding these constants, but I agree with @weierophinney in the original thread that they'd be better in their own interface.  So here 'tis.

I built this off of @dunglas's original branch so that he still gets a commit credit for it.

### CODE REVIEW

**Comment by evert** [comment]
Here's the official registry, tied to RFC8288: https://www.iana.org/assignments/link-relations/link-relations.xhtml

Here's another list that is referenced by other sources: http://microformats.org/wiki/existing-rel-values

Not sure what adding these constants add exactly, but now you can go totally nuts with them =)

**Comment by Crell** [comment]
Hm.  Yeah, the list has grown since I was last using it...

I'm a little torn.  On the one hand, if we can make constants for more standard relationships we should.  On the other, it's not a small list. :smile:

I'm tempted to say we should therefore break them up into separate interfaces by category or something, but that's a rabbit hold of trying to figure out which goes where, both for us and for users.  Thoughts from anyone else?

The goal of having the constants is simply to make typos into syntax errors for common relationships.  There's no functional difference; it's just a syntax helper for authors, and a shared index of popular/standard values.

**Comment by evert** [comment]
I don't really have a stake in this, but the IANA list is growing and will continue to grow. It feels a little odd to me to have something like this.

But if you are committed to this, I would probably suggest you at least auto-generate it based on the IANA list.

File: src/Link.php [hunk 1]
```diff
@@ -1,6 +1,5 @@
 <?php
 
-
```
Comment (weierophinney) [inline src/Link.php:3]: This is unrelated and should likely be done in a new PR.

File: src/Relationships.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<?php
+declare(strict_types=1);
+
+namespace Fig\Link;
+
+interface Relationships
+{
+    // Relations defined in https://www.w3.org/TR/html5/links.html#links and applicable on link elements
+    const REL_ALTERNATE = 'alternate';
```
Comment (weierophinney) [inline src/Relationships.php:9]: These should likely have `public` visibility declared, considering we target PHP 7 for this spec.

**Review by weierophinney — CHANGES_REQUESTED** [review]
I agree with @evert here - this is most useful when we have tools to autogenerate the relations, or at least check to ensure we have constants covering each relation in the IANA list.

File: src/Link.php (hunk 1 above)
Comment (Crell) [reply src/Link.php:3]: Accidental from moving code around between the two approaches, I think.  Or maybe someone's CS fixer along the way.

File: src/Relationships.php (hunk 2 above)
Comment (Crell) [reply src/Relationships.php:9]: Technically the current version targets PHP 5; it won't target 7 until the 1.1/2.0 type changes go through.  Visibility here would mean bumping the version requirement for the util package.

Probably good to do that with the next release that also uses the 2.0 version of the PSR.

**Comment by Crell** [comment]
Closing in favor of #6
